scripts/viewer/run_viewer.py

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/env python
+"""
+Starts viewer in eval mode.
+"""
+from __future__ import annotations
+
+import dataclasses
+import time
+from dataclasses import dataclass, fields
+from pathlib import Path
+
+import tyro
+from rich.console import Console
+
+from nerfstudio.configs.base_config import ViewerConfig
+from nerfstudio.engine.trainer import TrainerConfig
+from nerfstudio.pipelines.base_pipeline import Pipeline
+from nerfstudio.utils import profiler, writer
+from nerfstudio.utils.eval_utils import eval_setup
+from nerfstudio.utils.writer import EventName, TimeWriter
+from nerfstudio.viewer.server import viewer_utils
+
+CONSOLE = Console(width=120)
+
+
+@dataclass
+class ViewerConfigWithoutNumRays(ViewerConfig):
+    """Configuration for viewer instantiation"""
+
+    num_rays_per_chunk: tyro.conf.Suppress[int] = -1
+    start_train: tyro.conf.Suppress[bool] = False
+
+    def as_viewer_config(self):
+        """Converts the instance to ViewerConfig"""
+        return ViewerConfig(**{x.name: getattr(self, x.name) for x in fields(self)})
+
+
+@dataclass(frozen=True)
+class RunViewer:
+    """Load a checkpoint and start the viewer."""
+
+    load_config: Path
+    """Path to config YAML file."""
+    viewer: ViewerConfigWithoutNumRays = ViewerConfigWithoutNumRays()
+    """Viewer configuration"""
+
+    def main(self) -> None:
+        """Main function."""
+        config, pipeline, _ = eval_setup(
+            self.load_config,
+            eval_num_rays_per_chunk=None,
+            test_mode="test",
+        )
+        num_rays_per_chunk = config.viewer.num_rays_per_chunk
+        assert self.viewer.num_rays_per_chunk == -1
+        config.vis = "viewer"
+        config.viewer = self.viewer.as_viewer_config()
+        config.viewer.num_rays_per_chunk = num_rays_per_chunk
+
+        self._start_viewer(config, pipeline)
+
+    def _start_viewer(self, config: TrainerConfig, pipeline):
+        base_dir = config.get_base_dir()
+        viewer_log_path = base_dir / config.viewer.relative_log_filename
+        viewer_state, banner_messages = None, None
+        viewer_state, banner_messages = viewer_utils.setup_viewer(
+            config.viewer, log_filename=viewer_log_path, datapath=config.pipeline.datamanager.dataparser.data
+        )
+
+        # setup logging
+        writer.setup_local_writer(config.logging, max_iter=config.max_num_iterations, banner_messages=banner_messages)
+        writer.put_config(name="config", config_dict=dataclasses.asdict(config), step=0)
```

Comment:
Does it run without this?

Reply:
It does. Is it ok not to print the banner messages (f"Viewer at: {viewer_state.viewer_url}")? The viewer_url is already mentioned in the log, so I guess it is safe to remove.

Reply:
No, it doesn't work without the local_writer

Reply:
I still need `writer.setup_local_writer`. Otherwise I get the following error:
![image](https://user-images.githubusercontent.com/4222114/217623401-e8e50bec-d176-4c45-9b8c-a108b683d000.png)

Reply:
I will remove the dependency